### PR TITLE
pandoc-include: init at 1.2.0

### PR DESCRIPTION
**Pull request by ppenguin**
###### Description of changes

[`pandoc-include`](https://github.com/DCsunset/pandoc-include) is a python-based pandoc filter to allow file and header includes.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rnhmjoj** [comment]
I believe this should be considered an application, i.e. not intended to be imported and used in a python program.
If so, you should use `buildPythonApplication` instead and move the definition to `pkgs/top-level/all-packages.nix`.

This is what the manual says about it:

> The buildPythonApplication function is practically the same as buildPythonPackage. The main purpose of this function is to build a Python package where one is interested only in the executables, and not importable modules. For that reason, when adding this package to a python.buildEnv, the modules won’t be made available.

> All Python packages reside in pkgs/top-level/python-packages.nix and all applications elsewhere.

File: pkgs/tools/misc/pandoc-include/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  src = fetchFromGitHub {
+    owner = "DCsunset";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-kuxud7m+sWcNqE8A+Fwb8ATgiUwxQvHeYBTyw1UzX4U=";
+  };
+
+  propagatedBuildInputs = [ natsort panflute ];
+
+  doCheck = false;
```
Comment (rnhmjoj) [inline pkgs/tools/misc/pandoc-include/default.nix:22]: If the package has no check phase you don't need to disable it: Nix will simply skip it.
The repository does have tests, but need pandoc and latex which seems a bit excessive.
As a lightweight alternative you can do:
```suggestion
  pythonImportsCheck = [ "pandoc_include.main" ];
```

File: pkgs/tools/misc/pandoc-include/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  meta = with lib; {
+    description = "Pandoc filter to allow file and header includes.";
```
Comment (rnhmjoj) [inline pkgs/tools/misc/pandoc-include/default.nix:25]: The convention is to *not* have a final period in the description.

**Review by rnhmjoj — CHANGES_REQUESTED** [review]
A couple more suggestions, sorry If I didn't report them earlier.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 164552` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>pandoc-include</li>
</ul>
</details>

**Pull request merged**